homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,322 @@
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE_RANGE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_MODE,
+    ATTR_NAME,
+    CONF_COMMAND_OFF,
+    CONF_COMMAND_ON,
+    EVENT_HOMEASSISTANT_STOP,
+    PRECISION_TENTHS,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+import homeassistant.util.dt as dt_util
+
+from .const import (
+    ATTR_OVERRIDE_ALLOWED,
+    ATTR_TARGET_ID,
+    ATTR_TARGET_TYPE,
+    ATTR_TEMP_COMFORT_C,
+    ATTR_TEMP_ECO_C,
+    CONF_OVERRIDE_TYPE,
+    CONF_OVERRIDE_TYPE_NOW,
+    DOMAIN,
+)
+
+SUPPORT_FLAGS = SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE_RANGE
+
+PRESET_MODES = [PRESET_NONE, PRESET_COMFORT, PRESET_ECO, PRESET_AWAY]
+
+HVAC_MODES = [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_AUTO]
+HVAC_MODES_WITHOUT_OFF = [HVAC_MODE_HEAT, HVAC_MODE_AUTO]
+
+MIN_TEMPERATURE = 7
+MAX_TEMPERATURE = 40
+
+_LOGGER = logging.getLogger(__name__)
+
+_ZONE_NORMAL_WEEK_LIST_SCHEMA = vol.Schema({cv.string: cv.string})
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> bool:
+    """Set up the Nobø Ecohub platform from UI configuration."""
+
+    # Setup connection with hub
+    hub = hass.data[DOMAIN][config_entry.entry_id]
+
+    override_type = (
+        nobo.API.OVERRIDE_TYPE_NOW
+        if config_entry.options.get(CONF_OVERRIDE_TYPE) == CONF_OVERRIDE_TYPE_NOW
+        else nobo.API.OVERRIDE_TYPE_CONSTANT
+    )
+
+    # Find OFF command (week profile) to use for all zones:
+    command_off_name = config_entry.options.get(CONF_COMMAND_OFF)
+    command_on_by_id: dict[str, str] = {}  # By default, nothing can be turned on
+    if command_off_name is None or command_off_name == "":
+        _LOGGER.debug(
+            "Not possible to turn off (or on) any zone, because OFF week profile was not specified"
+        )
+        command_off_id = None
+    else:
+        command_off_id = _get_id_from_name(command_off_name, hub.week_profiles)
+        if command_off_id == "" or command_off_id is None:
+            _LOGGER.warning(
+                "Can not turn off (or on) any zone, because week profile '%s' was not found",
+                command_off_name,
+            )
+        else:
+            _LOGGER.debug(
+                "To turn off any heater, week profile %s '%s' will be used",
+                command_off_id,
+                command_off_name,
+            )
+
+            # Find ON command (week profile) for the different zones:
+            command_on_dict = config_entry.options.get(CONF_COMMAND_ON)
+            if command_on_dict is None or command_on_dict.keys().__len__ == 0:
```

Comment:
```suggestion
            if not command_on_dict:
```
I think this should handle both cases

Reply:
Thanks - fixed.